### ISSUE TITLE
Mistake made in merged PR #800

### ISSUE THREAD

**Issue by jamessspanggg** [reposense/RepoSense#828]
PR #800 ensures the getFiltered() method to be executed only once when page is reloaded.

**Mistake in PR:**
The `beforeUpdate()` lifecycle hook will only be called when there are changes to the DOM (which I previously thought that it was called when there are changes to the data instead).

<img width="726" alt="Screen Shot 2019-07-22 at 4 59 04 PM" src="https://user-images.githubusercontent.com/32864116/61619915-5dce6e80-aca2-11e9-97b2-f7af157e4628.png">

**Issue:** 
There will be no DOM changes when the page is initially loaded. Hence, when the user changes a control that calls `getFiltered()`, the method will be skipped since `canGetFiltered` is still false.

Illustration of the issue:
![error](https://user-images.githubusercontent.com/32864116/61620512-860a9d00-aca3-11e9-86fa-36f955b9f7a4.gif) 

The first change in `sort within groups by` from `⬆️ title` to `⬇️ title` does not change the order of ramp charts because `getFiltered()` is skipped as `canGetFiltered` is false. However, the change from `⬆️ title` to `⬇️ title` is a DOM change, which will call the `beforeUpdate()` hook and set `canGetFiltered` to be true.

The second change in `sort within groups by` from `⬇️ title` to `⬆️ contribution` does change the order of ramp charts accordingly as `canGetFiltered` is now true and `getFiltered()` is not skipped.

**Suggestion:** 
Revert the commit for PR #800.

Sorry for the inconvenience caused.

**Comment by jamessspanggg** [comment]
@eugenepeh @yong24s

**Comment by yong24s** [comment]
Does setting the boolean in `mounted` helps?

```
mounted() {
    this.canGetFiltered = true; // enable getFiltered() after first cycle of watcher updates
},
```

**Comment by jamessspanggg** [comment]
> Does setting the boolean in `mounted` helps?

Nope. Watcher updates happen after `mounted` and before `beforeUpdate`

**Comment by eugenepeh** [comment]
How about `ready()`?

**Comment by jamessspanggg** [comment]
> How about `ready()`?

`ready()` occurs before `mounted()` as well. Need to find an event that happens after `mounted`.

<img width="668" alt="Screen Shot 2019-07-22 at 7 37 46 PM" src="https://user-images.githubusercontent.com/32864116/61629852-a7c24f00-acb8-11e9-99d2-eece17a5f5e4.png">

**Comment by jamessspanggg** [comment]
@eugenepeh may I make a PR to revert the commit?

**Comment by eugenepeh** [comment]
>@eugenepeh may I make a PR to revert the commit?

Sure, if there is no solution as of now. Btw can you include add the calling sequence of vue engine into dev guide as well?

**Comment by jamessspanggg** [comment]
> Btw can you include add the calling sequence of vue engine into dev guide as well?

Sure, can it be within PR #809 ?

**Comment by eugenepeh** [comment]
>Sure, can it be within PR #809 ?

If it is best fit under the HTML Report section, otherwise would be better to create another PR for it.

**Comment by jamessspanggg** [comment]
> If it is best fit under the HTML Report section, otherwise would be better to create another PR for it.

Ok. Will create a separate PR as it doesn't fit under HTML Report section.
